### PR TITLE
new calibrator, clusterizer, cpu and rechits

### PR DESCRIPTION
**Pull request by VinInn**

### CODE REVIEW

**Comment by VinInn** [comment]
does it need some sort of rebase?

File: SimG4Core/Geometry/test/Cons_.cpp [hunk 1]
```diff
@@ -45,13 +45,23 @@ testCons::matched_g4_and_dd( void )
...
   CPPUNIT_ASSERT( abs(g4v - ddv) < numeric_limits<float>::epsilon());
   CPPUNIT_ASSERT( abs(g4v - ddsv) < numeric_limits<float>::epsilon());
+=======
+  std::cout << "\tg4 volume = " << g4v <<" cm3" << std::endl;
+  std::cout << "\tdd volume = " << ddv << " cm3" <<  std::endl;
+  std::cout << "\tDD Information: " << dds << " vol=" << ddsv << " cm3" << std::endl;
+
+  CPPUNIT_ASSERT( std::fabs(g4v - ddv) < precision_in_digits );
+  CPPUNIT_ASSERT( std::fabs(g4v - ddsv) < precision_in_digits );
+>>>>>>> VinInn/gpuGainCalibs
```
Comment (makortel) [inline SimG4Core/Geometry/test/Cons_.cpp]: Something went wrong with rebase/merge?

**Comment by fwyzard** [comment]
can you try 
```
git rebase CMSSW_10_1_X_Patatrack
```
?

**Comment by fwyzard** [comment]
Otherwise, if this is ready to go, I can try to take care of the rebasing myself.

**Comment by VinInn** [comment]
and of course rebase does not work

```
CPPUNIT_TEST_SUITE_REGISTRATION( testCons );
[innocent@vinzen0 src]$ git rebase -i  cms-patatrack/CMSSW_10_1_X_Patatrack 
Auto-merging EventFilter/SiPixelRawToDigi/test/runRawToDigi_GPU_phase1.py
CONFLICT (add/add): Merge conflict in EventFilter/SiPixelRawToDigi/test/runRawToDigi_GPU_phase1.py
Auto-merging EventFilter/SiPixelRawToDigi/plugins/RawToDigiGPU.h
CONFLICT (add/add): Merge conflict in EventFilter/SiPixelRawToDigi/plugins/RawToDigiGPU.h
Auto-merging EventFilter/SiPixelRawToDigi/plugins/RawToDigiGPU.cu
CONFLICT (add/add): Merge conflict in EventFilter/SiPixelRawToDigi/plugins/RawToDigiGPU.cu
Auto-merging EventFilter/SiPixelRawToDigi/plugins/EventInfoGPU.h
CONFLICT (add/add): Merge conflict in EventFilter/SiPixelRawToDigi/plugins/EventInfoGPU.h
Auto-merging EventFilter/SiPixelRawToDigi/plugins/BuildFile.xml
CONFLICT (content): Merge conflict in EventFilter/SiPixelRawToDigi/plugins/BuildFile.xml
error: could not apply af2b739f5c7... Implementation of the pixel raw to digi algorithm in CUDA

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".

Could not apply af2b739f5c7... Implementation of the pixel raw to digi algorithm in CUDA
```

**Comment by VinInn** [comment]
#11 is what I would like to merge....

**Comment by fwyzard** [comment]
superseded by #13

**Pull request closed without merging**